templates/web/base/admin/extra-metadata-form.html

```diff
@@ -0,0 +1,81 @@
+        <label>
+            [% loc('Code') %]
+            <input name="metadata[[% loop.index %]].code" data-field-name="code" type=text value="[% meta.code | html %]">
+        </label>
+
+        <div class="admin-hint"><p>[% loc('Whether the user is required to provide a value for this field.') %]</p></div>
+        <label>
+            [% loc('Required') %]
+            <input name="metadata[[% loop.index %]].required" data-field-name="required"  type=checkbox [% meta.required == 'true' ? 'checked' : '' %]>
+        </label>
+
+        <div class="admin-hint"><p>[% loc('If unticked the user won’t see an input field, just the ‘Description’ text. This should generally be ticked.') %]</p></div>
+        <label>
+            [% loc('Variable') %]
+            <input name="metadata[[% loop.index %]].variable" data-field-name="variable"  type=checkbox [% meta.variable == 'true' ? 'checked' : '' %]>
+        </label>
+
+        <div class="admin-hint"><p>[% loc('The field name as shown to the user on the report form.') %]</p></div>
+        <label>
+            [% loc('Description') %]
+            <input name="metadata[[% loop.index %]].description" data-field-name="description" type=text value="[% meta.description | html %]">
+        </label>
+
+        <div class="admin-hint"><p>[% loc('Some extra guidance text shown to the user alongside the field. The default template does not show this (<code>meta.datatype_description</code>), you must add it in <code>category_extras_fields.html</code>') %]</p></div>
+        <label>
+            [% loc('Hint') %]
+            <input name="metadata[[% loop.index %]].datatype_description" data-field-name="datatype_description" type=text value="[% meta.datatype_description | html %]">
+        </label>
+
+        <div class="admin-hint"><p>[% loc('The type of input field to show to the user. <strong>Text</strong> is a simple text field, <strong>List</strong> is a drop-down selection.') %]</p></div>
+        <label>
+            [% loc('Type') %]
+            <select name="metadata[[% loop.index %]].datatype" data-field-name="datatype" class="js-metadata-item-type">
+                <option value="string" [% meta.datatype == 'string' ? 'selected' : '' %]>[% loc('String') %]</option>
+                <option value="singlevaluelist" [% meta.datatype == 'singlevaluelist' ? 'selected' : '' %]>[% loc('List') %]</option>
+            </select>
+        </label>
+
+        <div class="js-metadata-options">
+            <div class="admin-hint"><p>[% loc('For each option, <strong>Key</strong> is the value which is stored in the database for that option and <strong>Name</strong> is the value displayed to the user.') %]</p></div>
+            [% loc('Options') %]<span class="js-hidden"> [% loc('(ignored if type is "String")') %]</span>
+            <ul>
+                [% outer_loop = loop %]
+                [% values = meta.exists('values') ? meta.values : [] %]
```

Comment:
I wonder if zmeta.values OR []` would be better here, in case values exists but is an empty string (though I guess that couldn't/shouldn't happen?)